oxd-gen-client/src/test/java/io/swagger/client/api/GetTokensByCodeTest.java

```diff
@@ -39,6 +49,86 @@ public void test(String opHost, String redirectUrl, String userId, String userSe
         refreshToken(tokensResponse, client, site);
     }
 
+    @Parameters({"opHost", "redirectUrl", "userId", "userSecret"})
+    @Test
+    public void testWithInvalidAuthorizationCode(String opHost, String redirectUrl, String userId, String userSecret) throws Exception{
+
+        final DevelopersApi client = Tester.api();
+
+        final RegisterSiteResponse site = RegisterSiteTest.registerSite(client, opHost, redirectUrl);
+
+        final String state = CoreUtils.secureRandomString();
+
+        final String authorizationStr = Tester.getAuthorization(site);
+
+        final String code = codeRequest(client, site.getOxdId(), userId, userSecret, state, CoreUtils.secureRandomString(), authorizationStr);
```

Comment:
we don't need to obtain valid code if we just want to call test method with invalid code ? This line must be removed.

Reply:
Got it, thanks